src/test/e2e/services/user-service.e2e.test.ts

```diff
@@ -159,7 +159,23 @@ test('should not be allowed to create existing user', async () => {
             { username: 'test', rootRole: adminRole.id },
             TEST_AUDIT_USER,
         ),
-    ).rejects.toThrow(Error);
+    ).rejects.toThrow('User already exists');
+});
+
+test('should not be allowed to create existing user in a concurrency situation', async () => {
```

Comment:
This test case makes `createUser` method to handle a concurrent problem. It uses `Promise` to make the concurrency.